### PR TITLE
android: Upgrade to `jni 0.21`

### PR DESCRIPTION
**Pull request by MarijnS95**
Cc @rib

### CODE REVIEW

File: src/imp/platform/android.rs [hunk 1]
```diff
@@ -12,42 +12,29 @@ impl From<jni::errors::Error> for AppDirsError {
...
     method: &str,
     has_string_argument: bool,
 ) -> Result<String, AppDirsError> {
     let dir = if has_string_argument {
         env.call_method(
-            *context,
+            context,
             method,
             "(Ljava/lang/String;)Ljava/io/File;",
-            &[jni::objects::JValue::Void],
```
Comment (rib) [inline src/imp/platform/android.rs:24]: oof, yeah that wasn't correct before :)

Comment (MarijnS95) [reply src/imp/platform/android.rs:24]: Yup the new validation caught it!

**Comment by rib** [comment]
yeah, those changes look good to me.

the clippy issues look completely unrelated to this, so seems fine to ignore those here.

**Comment by MarijnS95** [comment]
> The clippy issues look completely unrelated to this, so seems fine to ignore those here.

New Rust version, stricter (really: more complete/correct/accurate) `clippy` lints, for better or for worse :)

**Comment by kornelski** [comment]
Thank you

**Comment by MarijnS95** [comment]
Thank you @kornelski for the quick release!

Fwiw I see additional clippy changes, are you running a configuration locally that differs from the CI?

**Comment by kornelski** [comment]
I just run clippy manually and selectively apply what seems sensible

**Comment by MarijnS95** [comment]
Would be great to list those in the CI and/or `lib.rs` so that contributors can proactively address them too.

**Comment by kornelski** [comment]
I'd rather not add stronger clippy requirements in the CI, because these are suggestions, not absolute requirements. If the clippy build in CI is confusing or getting in the way of PRs, I'd rather remove it from CI.

**Comment by MarijnS95** [comment]
Just asking how/where you'd manage to find more: manually requesting them on the command line or running a newer-than-stable toolchain?

**Comment by kornelski** [comment]
I've added `-W clippy::pedantic` to the flags.

**Pull request merged**